lib/motor.js

```diff
@@ -497,19 +517,32 @@ function Motor(opts) {
 
   // Derive device based on pins passed
   if (typeof this.opts.device === "undefined") {
-
-    this.opts.device = (typeof this.opts.pins === "undefined" && typeof this.opts.register !== "object") ?
-      "NONDIRECTIONAL" : "DIRECTIONAL";
-
-    if (this.opts.pins && (this.opts.pins.cdir || this.opts.pins.length > 2)) {
-      this.opts.device = "CDIR";
+    this.opts.device = "DIRECTIONAL";
```

Comment:
`Motor` is the only oddball class that does this thing with `this.opts` :|     
```js
Board.Component.call(
  this, this.opts = Board.Options(opts)
);
```

Every other class just does 

```js
Board.Component.call(
  this, opts = Board.Options(opts)
);
```
Because `Board.Component` and `Board.Options` normalize all the pins and put them directly on `this`. It works, so I wouldn't say change it all here and now, but I think we should get it consistent eventually.

Reply:
I bet there are some things kept on this.opts that I refer back to outside of the constructor. I just need to move that/those specific cases to the private state, or properties on this (depending on what it is).

Reply:
I'm going to handle this in a separate PR. I can just change the pins getter/setter on each device type to a normalizePins method that gets called by the constructor, receives opts as a param and returns the normalized pins object. The constructor will assign the returned value to this.pins.